File: app/src/main/java/com/glucocontrol/domain/usecase/auth/GetCurrentUserUseCase.kt
package com.glucocontrol.domain.usecase.auth

import com.glucocontrol.domain.model.User
import com.glucocontrol.domain.repository.AuthRepository
import javax.inject.Inject

class GetCurrentUserUseCase @Inject constructor(private val authRepository: AuthRepository) {
    suspend operator fun invoke(): User? = authRepository.getCurrentUser()
}

File: lib/domain/usecase/auth/get_current_user_use_case.dart
import 'package:glucocontrol/domain/model/user.dart';
import 'package:glucocontrol/domain/repository/auth_repository.dart';

class GetCurrentUserUseCase {
  final AuthRepository _repo;
  GetCurrentUserUseCase(this._repo);
  Future<AppUser?> call() => _repo.getCurrentUser();
}
